src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -137,14 +137,28 @@ public static async Task<RenameDocumentActionSet> RenameDocumentAsync(
 
     /// <inheritdoc cref="LightweightRenameLocations.FindRenameLocationsAsync"/>
     internal static Task<LightweightRenameLocations> FindRenameLocationsAsync(Solution solution, ISymbol symbol, SymbolRenameOptions options, CancellationToken cancellationToken)
-        => LightweightRenameLocations.FindRenameLocationsAsync(symbol, solution, options, cancellationToken);
+        => FindRenameLocationsAsync(solution, symbol, options, allowRenameInGeneratedDocument: false, cancellationToken);
+
+    /// <inheritdoc cref="LightweightRenameLocations.FindRenameLocationsAsync"/>
+    internal static Task<LightweightRenameLocations> FindRenameLocationsAsync(Solution solution, ISymbol symbol, SymbolRenameOptions options, bool allowRenameInGeneratedDocument, CancellationToken cancellationToken)
```

Comment:
why is allowRenameInGeneratedDocument not part of SymbolRenameOptions.  it morally feels like that options type exists just for this purpose.

Reply:
It would make sense, but it would also make it a public option (SymbolRenameOptions is public API). Are you okay with that?

Reply:
Yeah, I can make that work.